include/fs/fs.h

```diff
@@ -151,6 +152,8 @@ struct fs_statvfs {
  * @param mkdir Creates a new directory using specified path
  * @param stat Checks the status of a file or directory specified by the path
  * @param statvfs Returns the total and available space in the filesystem volume
+ * @param get_fs_data_storage Returns pointer to buffer for file system specific
```

Comment:
This commit `fs: Add internal API support for default fs storage allocator` needs a little more justification.  At this point I don't understand what it's for.